deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java

```diff
@@ -230,16 +235,70 @@ private Optional<String> getInputBaseDirRelativeToModule(final Path sourceDir, f
 
     private <T> Optional<T> getValues(final Config config, final Path openApiFilePath, CodegenConfig.ConfigName configName,
             Class<T> propertyType) {
-        return config
-                .getOptionalValue(CodegenConfig.getSpecConfigName(configName, openApiFilePath), propertyType)
-                .or(() -> config.getOptionalValue(CodegenConfig.getGlobalConfigName(configName), propertyType));
+
+        Optional<String> possibleConfigKey = getConfigKeyValue(config, openApiFilePath);
+        boolean userConfiguredConfigKeyProperty = possibleConfigKey.isPresent() && !possibleConfigKey.get().isBlank();
+
+        if (userConfiguredConfigKeyProperty) {
+            return getValuesByConfigKey(config, CodegenConfig.getSpecConfigNameByConfigKey(possibleConfigKey.get(), configName),
+                    propertyType, configName);
+        }
+
+        return getValuesBySpecConfigName(config, openApiFilePath, configName, propertyType);
     }
 
     private <K, V> Optional<Map<K, V>> getValues(final SmallRyeConfig config, final Path openApiFilePath,
             CodegenConfig.ConfigName configName,
             Class<K> kClass, Class<V> vClass) {
+
+        Optional<String> possibleConfigKey = getConfigKeyValue(config, openApiFilePath);
+        boolean userConfiguredConfigKey = possibleConfigKey.isPresent() && !possibleConfigKey.get().isBlank();
```

Comment:
I think you can isolate these lines into a function since it's the same as 239-240.

Reply:
I think that I can remove just one line, If I want to make this method more small will be necessary to create another two methods... because of the overloading to use `Class<K> kClass, Class<V> vClass` variation.